homeassistant/components/assist_satellite/entity.py

```diff
@@ -0,0 +1,72 @@
+"""Assist satellite entity."""
+
+from abc import abstractmethod
+from collections.abc import AsyncIterable
+
+from homeassistant.components.assist_pipeline import (
+    PipelineEventCallback,
+    PipelineStage,
+)
+from homeassistant.const import EntityCategory
+from homeassistant.core import Context
+from homeassistant.helpers import entity
+from homeassistant.helpers.entity import EntityDescription
+
+from .models import AssistSatelliteEntityFeature, AssistSatelliteState, SatelliteConfig
+
+
+class AssistSatelliteEntity(entity.Entity):
+    """Entity encapsulating the state and functionality of an Assist satellite."""
+
+    entity_description = EntityDescription(
+        key="assist_satellite",
+        translation_key="assist_satellite",
+        entity_category=EntityCategory.CONFIG,
+    )
+    _attr_should_poll = False
+    _attr_state: AssistSatelliteState | None = None
+    _attr_supported_features: AssistSatelliteEntityFeature = (
+        AssistSatelliteEntityFeature(0)
+    )
+
+    def __init__(self) -> None:
```

Comment:
Better not put default init on a class. You can set this attribute also at top level since it's static ?

Reply:
Wouldn't that cause the `SatelliteConfig` object to be shared among all instances then?

Reply:
I thought it's a frozen object that each implementation would override as necessary